[C++][cpp-restsdk] Simple request / response (string) is not well handled. Compilation failed

##### Description

A simple query with a simple response (string) does not compile.
The generated code shows a miscalled method that does not exist (std::string::fromJson) 

The compilation error looks like :
```
/home/xxxx/dev/amf/impl-ms1/src/client/client-ms2/gen-cpp/api/Stair1Api.cpp: In lambda function:
/home/xxxx/dev/amf/impl-ms1/src/client/client-ms2/gen-cpp/api/Stair1Api.cpp:161:27: error: base operand of ‘->’ has non-pointer type ‘utility::string_t {aka std::__cxx11::basic_string<char>}’
             localVarResult->fromJson(localVarJson);
                           ^
CMakeFiles/client-ms2.dir/build.make:254: recipe for target 'CMakeFiles/client-ms2.dir/api/Stair1Api.cpp.o' failed
make[3]: *** [CMakeFiles/client-ms2.dir/api/Stair1Api.cpp.o] Error 1

```

##### openapi-generator version

Current master 3.3.0-SNAPSHOT

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0
info:
  version: 1.0.0
  title: Check generation of  (cpp-restsdk)
  description: Internal ref filename is check_cpprest_simple_response.yaml 

servers:
  - url: http://localhost:8080

paths:
  /stair1:
    post:
      summary: blabla
      operationId: check_generation
      tags:
        - Stair1
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Content'
        required: true
      responses:
        '200':
          description: Expected response to a valid request
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Gpsi'
        default:
          description: Unexpected error

components:
  schemas:
    Content:
      type: object
      properties:
        data:
          $ref: '#/components/schemas/Gpsi'

    Gpsi:
      type: string
```

##### Command line used for generation

Generate :
```
openapi-generator-cli.sh generate -i ./openapi.yaml -g cpp-restsdk -c ./config.json -o gen-cpp
```

Compile :
```
mkdir -p obj ; cd obj ; cmake -D CPPREST_ROOT=/usr/ ../gen-cpp ; make ; cd ..
```

##### Steps to reproduce

Generate & Compile

##### Related issues/PRs

##### Suggest a fix/enhancement

Wrong code is located in _Stair1Api::checkGeneration_ method :

```
../..
    .then([=](utility::string_t localVarResponse)
    {
        utility::string_t localVarResult(utility::conversions::to_string_t(""));

        if(localVarResponseHttpContentType == utility::conversions::to_string_t("application/json"))
        {
            web::json::value localVarJson = web::json::value::parse(localVarResponse);

            // WRONG CALL BELOW
            localVarResult->fromJson(localVarJson);
        }
        // else if(localVarResponseHttpContentType == utility::conversions::to_string_t("multipart/form-data"))
        // {
        // TODO multipart response parsing
        // }
        else
        {
            throw ApiException(500
                , utility::conversions::to_string_t("error calling checkGeneration: unsupported response type"));
        }

        return localVarResult;
    });
../..
```

The called method should probably be :

```
    localVarResult = ModelBase::stringFromJson(localVarJson);
```

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestSdkClientCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestSdkClientCodegen.java
@@ -38,6 +38,7 @@ import java.util.Arrays;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
@@ -154,7 +155,7 @@ public class CppRestSdkClientCodegen extends AbstractCppCodegen {
         typeMapping.put("map", "std::map");
         typeMapping.put("file", "HttpContent");
         typeMapping.put("object", "Object");
-        typeMapping.put("binary", "std::string");
+        typeMapping.put("binary", "utility::string_t");
         typeMapping.put("number", "double");
         typeMapping.put("UUID", "utility::string_t");
         typeMapping.put("ByteArray", "utility::string_t");
@@ -249,6 +250,7 @@ public class CppRestSdkClientCodegen extends AbstractCppCodegen {
 
             if (methodResponse != null) {
                 Schema response = ModelUtils.getSchemaFromResponse(methodResponse);
+                response = ModelUtils.unaliasSchema(openAPI.getComponents().getSchemas(), response);
                 if (response != null) {
                     CodegenProperty cm = fromProperty("response", response);
                     op.vendorExtensions.put("x-codegen-response", cm);
@@ -276,6 +278,24 @@ public class CppRestSdkClientCodegen extends AbstractCppCodegen {
         }
     }
 
+    // override with any special post-processing
+    @SuppressWarnings("unchecked")
+    @Override
+    public Map<String, Object> postProcessOperationsWithModels(Map<String, Object> objs, List<Object> allModels) {
+        Map<String, Object> operations = (Map<String, Object>) objs.get("operations");
+        List<CodegenOperation> operationList = (List<CodegenOperation>) operations.get("operation");
+        for (CodegenOperation op : operationList) {
+            for(String hdr : op.imports) {
+                if(importMapping.containsKey(hdr)) {
+                    continue;
+                }
+                operations.put("hasModelImport", true);
+                break;
+            }
+        }
+        return objs;
+    }
+    
     protected boolean isFileSchema(CodegenProperty property) {
         return property.baseType.equals("HttpContent");
     }

--- a/modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/api-header.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/api-header.mustache
@@ -13,6 +13,7 @@
 
 {{#imports}}{{{import}}}
 {{/imports}}
+{{^hasModelImport}}#include "../ModelBase.h"{{/hasModelImport}}
 
 #include <boost/optional.hpp>
 

--- a/samples/client/petstore/cpp-restsdk/api/PetApi.h
+++ b/samples/client/petstore/cpp-restsdk/api/PetApi.h
@@ -26,6 +26,7 @@
 #include "Pet.h"
 #include <cpprest/details/basic_types.h>
 
+
 #include <boost/optional.hpp>
 
 namespace org {

--- a/samples/client/petstore/cpp-restsdk/api/StoreApi.h
+++ b/samples/client/petstore/cpp-restsdk/api/StoreApi.h
@@ -25,6 +25,7 @@
 #include <map>
 #include <cpprest/details/basic_types.h>
 
+
 #include <boost/optional.hpp>
 
 namespace org {

--- a/samples/client/petstore/cpp-restsdk/api/UserApi.h
+++ b/samples/client/petstore/cpp-restsdk/api/UserApi.h
@@ -25,6 +25,7 @@
 #include <vector>
 #include <cpprest/details/basic_types.h>
 
+
 #include <boost/optional.hpp>
 
 namespace org {

